client/data/posts/use-create-new-post.js

```diff
@@ -0,0 +1,28 @@
+import { useCallback } from 'react';
+import { useMutation } from 'react-query';
```

Comment:
Nice to see React Query used for post creation!

Note that we might need to change the import statement since we're in the middle of upgrading the package to v4 in #74381.

Reply:
Thanks for flagging the react-query update!